File: src/java/magmac/app/lang/java/Segment.java
package magmac.app.lang.java;

import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.Node;

public record Segment(String value) {
    public static CompileResult<Segment> deserialize(Node node) {
        return node.deserialize()
                .withString("value")
                .complete(Segment::new);
    }
}

File: src/web/magmac/app/lang/node/Segment.ts
export class Segment {
	public static deserialize( node : Node) : CompileResult<Segment> {return Destructors.destruct( node).withString( "value").complete( Segment.new);;}
	public serialize() : Node {return new MapNode( "segment").withString( "value", this.value);;}
}
